db/version_set.cc

```diff
@@ -2132,7 +2132,8 @@ void Version::Get(const ReadOptions& read_options, const LookupKey& k,
     }
     // merge_operands are in saver and we hit the beginning of the key history
     // do a final merge of nullptr and operands;
-    std::string* str_value = value != nullptr ? value->GetSelf() : nullptr;
+    std::string tmp_value;
+    std::string* str_value = value != nullptr ? value->GetSelf() : &tmp_value;
```

Comment:
Can you explain this one to me?  I am not sure I understand what the error is/was and how this addresses it.  This appears to make it so that nullptr cannot/will not be passed into TimedFullMerge, which I am guessing can change the semantics of that method

Reply:
Coverity report a FORWARD_NULL error:  `MergeHelper::TimedFullMerge` deference `str_value` pointer with no NULL check，in db/merge_helper.cc:78
```cpp
MergeOperator::MergeOperationOutput merge_out(*result, tmp_result_operand);
```

Reply:
I added a NULL assert in  db/merge_helper.cc:78
```cpp
  assert(result != nullptr);
```
maybe this looks more clear.